### PR TITLE
Update UI and Styling for Contact Management App

### PR DESCRIPTION
**Pull request by mawalou14**
Closes: #823

- **Title:** Update UI and Styling for Contact Management App
- **Name:** Mouhamadou Awalou
- **Idenitfy yourself:** Contributor in Hacktoberfest

### Describe the add-ons or changes you've made 📃
In this contribution, I have updated the UI and styling of the Contact Management App located in the advanced section of the Angular-js projects. The changes include improved visual elements for better user experience, including a responsive layout, updated color schemes, and enhanced component styling.
Also,  updated the readme.md file of this project by replacing the screenshots with the new look.

### Screenshots 📷
contact management app
noContactFound
successPopup

### Type of change ☑️
What sort of change have you made:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Checklist: ☑️
- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CONTRIBUTING.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Note to reviewers 📄
Please review the UI changes and provide feedback on the overall design and usability improvements. I appreciate your input!. The background image used is a free picture downloaded from freepik.com.

### CODE REVIEW

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
Delete all the `.angular` cache files and other any un-necessary files.

**Comment by Avdhesh-Varshney** [comment]
Can you tell what you have improved except bg addition?

**Comment by mawalou14** [comment]
> Can you tell what you have improved except bg addition?

I've made serceral improvements beyond just adding the background image. Here are the changes and improvements I made:

- [x] Added toastr notifications for all operations ( Add, Update, Delete).
- [x] Updated the subscription syntax to follow best practices.
- [x]  Removed unnecessary full-page reloads by notifying the contact list component via the contact service, ensuring smooth updates.
- [x]  Added meaningful comments to help other developers understand the changes.
- [x]  Improved UI layout by making the header fixed and placing the search bar in the header.
- [x] Added a "No contact found" message when the contact list is empty.
- [x] Updated the README.md for better documentation and understanding.

**Comment by mawalou14** [comment]
> Delete all the `.angular` cache files and other any un-necessary files. 

Hi @Avdhesh-Varshney. Done, I've pushed the requested changes, and all feedback has been implemented.

File: Angular-JS-Projects/Advanced/Contact-Manager/screenshot.webp
Comment (Avdhesh-Varshney) [inline Angular-JS-Projects/Advanced/Contact-Manager/screenshot.webp:1]: Undo this image. `screenshot.webp` file is mandatory in every project.

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]

File: Angular-JS-Projects/Advanced/Contact-Manager/screenshot.webp
Comment (mawalou14) [reply Angular-JS-Projects/Advanced/Contact-Manager/screenshot.webp:1]: Hi, @Avdhesh-Varshney, Done I restored the mandatory screenshot.webp file

File: Angular-JS-Projects/Advanced/Contact-Manager/README.md
Comment (Avdhesh-Varshney) [inline Angular-JS-Projects/Advanced/Contact-Manager/README.md:1]: No screenshot file. No image is there to render in readme file?
image

Comment (Avdhesh-Varshney) [reply Angular-JS-Projects/Advanced/Contact-Manager/README.md:1]: You should also review the readme file and show me also

Comment (mawalou14) [reply Angular-JS-Projects/Advanced/Contact-Manager/README.md:1]: Alright, no worries

Comment (mawalou14) [reply Angular-JS-Projects/Advanced/Contact-Manager/README.md:1]: > You should also review the readme file and show me also

Hi, @Avdhesh-Varshney, the screenshots (05 in number) have been updated, then added all those to the readme file for inorder to ease the understanding of the project UI. 
So now the readme file is updated and well documented. Below is the a screenshot of part of the readme file.

Screenshot 2024-10-16 223615

**Review by Avdhesh-Varshney — APPROVED** [review]

**Pull request merged**